### PR TITLE
Fix: Order of VoDs Is Reversed When Not Logged-In

### LINKED ISSUE #485: Order of VoDs Is Reversed When Not Logged-In
As a follow-up to #447 and in particular https://github.com/joschahenningsen/TUM-Live/pull/447#issuecomment-1124331674.

When no user is logged-in, the course overview has the VoD list in the reversed order (the "DESC" order puts the newest lectures at the bottom instead of the top).

This is because adding the watch state reverses the sort sort order w.r.t. the initial query:
https://github.com/joschahenningsen/TUM-Live/blob/c0869dd2c6d2fbfaae021699455232473464a0a5/dao/streams.go#L277
https://github.com/joschahenningsen/TUM-Live/blob/c0869dd2c6d2fbfaae021699455232473464a0a5/dao/courses.go#L222

### PR DESCRIPTION
**Pull request by just-max**
Fixes #485

### CODE REVIEW

File: web/template/vod_course_list.gohtml [hunk 1]
```diff
@@ -96,7 +96,7 @@
             </li>
         {{end}}
     {{end}}
-    {{if and .User $course.HasRecordings}}
+    {{if and $user $course.HasRecordings}}
```
Comment (alexanderstephan) [inline web/template/vod_course_list.gohtml:99]: Good catch, can you also remove the corresponding `User` field from `CoursePageData`? I don't think it's used anywhere else.

**Review by alexanderstephan — CHANGES_REQUESTED** [review]
Nice fix, just some minor note that could be done before merge.

File: web/template/vod_course_list.gohtml (hunk 1 above)
Comment (just-max) [reply web/template/vod_course_list.gohtml:99]: Removed.

**Review by alexanderstephan — APPROVED** [review]
Thanks! 👍

**Pull request merged**